### ISSUE TITLE
Possible to support displaying all `authors` on PyPI page?

### ISSUE THREAD

**Issue by matthewfeickert** [pypi/warehouse#12877]
**What's the problem this feature will solve?**

:wave: Hi. Our project [`pyhf`](https://github.com/scikit-hep/pyhf) just switched (c.f. https://github.com/scikit-hep/pyhf/pull/2095) from having our PyPI metadata in [`setup.cfg`](https://github.com/scikit-hep/pyhf/blob/3eef1fff2e6d4ccf474bbe51e2211cf690752b82/setup.cfg) to [`pyproject.toml`](https://github.com/scikit-hep/pyhf/blob/e942d852922f22c495ae711303835a5a698f217d/pyproject.toml). In doing so, we also changed from having our author metadata for the 3 authors be across [`author` and `author_email`](https://github.com/scikit-hep/pyhf/blob/3eef1fff2e6d4ccf474bbe51e2211cf690752b82/setup.cfg#L7-L8) to having it be contained in [`authors`](https://github.com/scikit-hep/pyhf/blob/e942d852922f22c495ae711303835a5a698f217d/pyproject.toml#L12-L16) following [PEP 621's requirements](https://peps.python.org/pep-0621/#authors-maintainers) of

> These fields accept an array of tables with 2 keys: name and email. Both values must be strings. The name value MUST be a valid email name (i.e. whatever can be put as a name, before an email, in [RFC 822](https://datatracker.ietf.org/doc/html/rfc822.html)) and not contain commas. The email value MUST be a valid email address. Both keys are optional.

`pip` is recognizing all the metadata as we would expect

```console
$ python -m pip show pyhf
Name: pyhf
Version: 0.7.1.dev43
Summary: pure-Python HistFactory implementation with tensors and autodiff
Home-page: 
Author: 
Author-email: Lukas Heinrich <lukas.heinrich@cern.ch>, Matthew Feickert <matthew.feickert@cern.ch>, Giordon Stark <gstark@cern.ch>
License: Apache-2.0
Location: /home/feickert/.pyenv/versions/3.10.6/envs/pyhf-dev-CPU/lib/python3.10/site-packages
Requires: click, jsonpatch, jsonschema, numpy, pyyaml, scipy, tqdm
Required-by:
```

However, when we [published this to TestPyPI](https://test.pypi.org/project/pyhf/0.7.1.dev35/) to check how things looked after switching over we noticed that TestPyPI is displaying only the first author and linking their email

[![testPyPI](https://user-images.githubusercontent.com/5142394/213782977-8b308852-5f24-4d9f-a946-2fb00af5c301.png)](https://test.pypi.org/project/pyhf/0.7.1.dev35/)

Previously when we shoved all our names and emails into `author` and `author_email` we could at least have [all our names be displayed](https://pypi.org/project/pyhf/0.7.0/) (no surprise there as we were abusing the field)

[![PyPI-0 7 0](https://user-images.githubusercontent.com/5142394/213784164-d3cfc0af-f5d7-4037-a88c-2e8fcda37aac.png)](https://pypi.org/project/pyhf/0.7.0/)

I assume that this behavior with `authors` is because `warehouse` uses only the core metadata here (?) following PEP 621's instructions of:

> Using the data to fill in [core metadata](https://packaging.python.org/specifications/core-metadata/) is as follows:
> 
> 1. If only name is provided, the value goes in Author/Maintainer as appropriate.
> 2. If only email is provided, the value goes in Author-email/Maintainer-email as appropriate.
> 3. If both email and name are provided, the value goes in Author-email/Maintainer-email as appropriate, with the format {name} <{email}> (with appropriate quoting, e.g. using email.headerregistry.Address).
> 4. Multiple values should be separated by commas.

Would it be possible for `warehouse` to display all authors information if it exists? Or is that something that is outside the scope of how `warehouse` interacts with metadata?

**Describe the solution you'd like**

Have `warehouse` be able to parse the existence of PEP 621 `authors` and display all names and associated emails of `authors` on the package webpage.

**Additional context**

**Comment by di** [comment]
Is this a duplicate of #9400?

**Comment by matthewfeickert** [comment]
> Is this a duplicate of #9400?

Hm I think so. I'll assume yes and close here in favor of it, but actually check when I'm out of my next meeting.

Apologies for the noise: I searched "authors" in the Issues and so missed this as I didn't check all the ones just matching "author".

**Comment by di** [comment]
Reopening because this is distinct from #9400 per https://github.com/pypi/warehouse/issues/9400#issuecomment-2318213129.

**Comment by nbehrnd** [comment]
Departing from the [Python Packaging Guide] I'm getting more familiar with this indicator on test.pypi.org.  I would like to use a `pyproject.toml` about two authors, one to serve as corresponding author with an email, i.e.

``` toml
authors = [
  { name="Page Book", email="test@example.com" },
  { name="Pen Paper" },
]

maintainers = [
  { name="Norwid Behrnd", email="nbehrnd@yahoo.com" },
]
``` 

While acceptable to a build by `uv build`, test.pypi.org only displays the last author while the envelope icon links to the address of the first author:

<img width="928" height="559" alt="Image" src="https://github.com/user-attachments/assets/04f07dab-7ce6-4fdd-82d5-91e3276d1903" />

[Python Packaging Guide]: https://packaging.python.org/en/latest/guides/writing-pyproject-toml/

I get to get both authors back on display if none of them is described with an email:

<img width="987" height="620" alt="Image" src="https://github.com/user-attachments/assets/41bb23b8-6c01-407c-b5f9-733bbc5e1c37" />
